javaslang/src/test/java/javaslang/collection/TreeMapTest.java

```diff
@@ -122,6 +126,22 @@ public void shouldWrapMap() {
         assertThat(TreeMap.ofAll(source)).isEqualTo(emptyIntInt().put(1, 2).put(3, 4));
     }
 
+    // -- ofAll
+
+    @Test
+    public void shouldCreateKeyComparatorForJavaUtilMap() {
+        final TreeMap<String, Integer> actual = TreeMap.ofAll(mapOfTuples(Tuple.of("c", 0), Tuple.of("a", 0), Tuple.of("b", 0)).toJavaMap());
+        final List<String> expected = List.of("a", "b", "c");
+        assertThat(actual.keySet().toList()).isEqualTo(expected);
```

Comment:
All right. It passed before. Just removed `mapOfTuples` because it creates a TreeMap to create a TreeMap. But here it does not really matter.